jetty-util/src/main/java/org/eclipse/jetty/util/RolloverFileOutputStream.java

```diff
@@ -335,35 +334,53 @@ void removeOldFiles(ZonedDateTime now)
     }
 
     /* ------------------------------------------------------------ */
+    public void write(int b) throws IOException
+    {
+        _out.write(b);
+    }
+
+    /* ------------------------------------------------------------ */
     @Override
     public void write (byte[] buf)
         throws IOException
     {
-        out.write (buf);
+        _out.write (buf);
     }
 
     /* ------------------------------------------------------------ */
     @Override
     public void write (byte[] buf, int off, int len)
         throws IOException
     {
-        out.write (buf, off, len);
+        _out.write (buf, off, len);
     }
 
     /* ------------------------------------------------------------ */
+    public void flush() throws IOException
+    {
+        _out.flush();
+    }
+    
+    /* ------------------------------------------------------------ */
     @Override
     public void close()
         throws IOException
     {
-        synchronized(RolloverFileOutputStream.class)
+        synchronized(this)
         {
-            try{super.close();}
+            try
+            {
+                _out.close();
+            }
             finally
             {
-                out=null;
+                _out=null;
                 _file=null;
             }
-    
+        }
+
+        synchronized(RolloverFileOutputStream.class)
```

Comment:
Is synchronized(RolloverFileOutputStream.class) here and above you made it synchronized(this), is that right?

Reply:
The class muted is used only when accessing the shared static timer, which is done by the cancel call